bliss/surveys/dc2.py

```diff
@@ -262,33 +382,43 @@ def from_file(cls, cat_path, wcs, height, width, band):
             "star_log_fluxes": star_log_fluxes.reshape(1, nobj, 6),
         }
 
-        return cls(height, width, d), torch.stack(psf_params)
+        if return_match_id:
+            return cls(height, width, d), torch.stack(psf_params), match_id
+        else:
+            return cls(height, width, d), torch.stack(psf_params)
 
 
 def read_frame_for_band(image_files, bg_files, n, n_bands, image_lim):
     image_list = []
     bg_list = []
     wcs = None
+    wcs_header_str = None
     for b in range(n_bands):
         image_frame = fits.open(image_files[b][n])
         bg_frame = fits.open(bg_files[b][n])
         image_data = image_frame[1].data  # pylint: disable=maybe-no-member
         bg_data = bg_frame[0].data  # pylint: disable=maybe-no-member
 
         if wcs is None:
-            wcs = WCS(image_frame[1].header)  # pylint: disable=maybe-no-member
+            wcs_header = image_frame[1].header
+            wcs = WCS(wcs_header)  # pylint: disable=maybe-no-member
+            wcs_header_str = wcs_header.tostring()
 
         image_frame.close()
         bg_frame.close()
 
         image = torch.nan_to_num(torch.from_numpy(image_data)[: image_lim[0], : image_lim[1]])
         bg = torch.from_numpy(bg_data.astype(np.float32)).expand(image_lim[0], image_lim[1])
 
+        # to make sure the background are positive
+        bg = bg.clone()
+        bg[bg <= 1e-6] = 1e-5
```

Comment:
Probably better to leave these unchanged. If they are negative and we run code that is expecting positive values, it would be better to have the code raise an error.

Reply:
Delete it in commit a521a09.